docs/contributing/Building, Debugging, and Testing on Windows.md

```diff
@@ -0,0 +1,91 @@
+# Required Software
+
+1. [Visual Studio 2015 with Update 1](http://go.microsoft.com/fwlink/?LinkId=691129). _You need Update 1_.
+2. Visual Studio 2015 Extensibility Tools. If you already installed Visual Studio, choose "Modify" from the Programs and Features control panel, and check "Visual Studio Extensibility".
+
+# Getting the Code
+
+1. Clone https://github.com/dotnet/roslyn
+2. Run the "Developer Command Prompt for VS2015" from your start menu.
+3. Run `Restore.cmd` in the command prompt to restore NuGet packages.
+4. Due to [Issue #5876](https://github.com/dotnet/roslyn/issues/5876), you should build on the command line before opening in Visual Studio. Run `msbuild /v:m /m Roslyn.sln`
```

Comment:
Again let me clarify, I'm suggesting documenting.

@jasonmalinowski 1. Yes, I agree; 2. Yes, that's what I've been doing

@jaredpar 1. I remember this BootrapBuildPath property was documented in a section like "Using the OSS built compiler", but not sure if it also mentioned toolset.sln; 2. Yes, but worth doing locally before submitting pr with compiler changes, or would probably cause unnecessary trouble to the infrastructure team like I did in #6508

Well, thanks for all these info of the discussion, I think it's pretty clear to me, maybe I can try putting this into a new section tomorrow.